testing/tests/closures.rs

```diff
@@ -0,0 +1,79 @@
+use askama::Template;
+
+const FALSE: &bool = &false;
+
+#[derive(Debug, Clone)]
+struct User {
+    name: String,
+    flag: bool,
+}
+
+impl User {
+    fn ferris() -> Self {
+        Self {
+            name: "Ferris".to_string(),
+            flag: true,
+        }
+    }
+}
+
+#[derive(Template)]
+#[template(
+    source = r#"Hello {{ user_opt.map(|user| user.name.as_str()).unwrap_or("World") }}"#,
+    ext = "txt"
+)]
+struct ClosureTemplate<'a> {
+    user_opt: Option<&'a User>,
+}
+
+#[test]
+fn test_closure() {
+    let user = User::ferris();
+    let t = ClosureTemplate {
+        user_opt: Some(&user),
+    };
+    assert_eq!(t.render().unwrap(), "Hello Ferris");
+
+    let t = ClosureTemplate { user_opt: None };
+    assert_eq!(t.render().unwrap(), "Hello World");
+}
+
+#[derive(Template)]
+#[template(
+    source = r#"Hello {{ user.map(|user| user.name.as_str()).unwrap_or("World") }}"#,
+    ext = "txt"
+)]
+struct ClosureShadowTemplate<'a> {
+    user: Option<&'a User>,
+}
+
+#[test]
+fn test_closure_shadow() {
+    let user = User::ferris();
+    let t = ClosureShadowTemplate { user: Some(&user) };
+    assert_eq!(t.render().unwrap(), "Hello Ferris");
+
+    let t = ClosureShadowTemplate { user: None };
+    assert_eq!(t.render().unwrap(), "Hello World");
+}
+
+#[derive(Template)]
+#[template(
+    source = r#"{{ user_opt.map(|user| user.flag).unwrap_or(FALSE) }}"#,
```

Comment:
I understand why `FALSE` is needed, but it's still possible to do the following, right?

```suggestion
    source = r#"{{ user_opt.map(|user| user.flag).copied().unwrap_or(false) }}"#,
```

Reply:
Yes, that should work too. I used `FALSE` to statically ensure that there really is a reference. But since `copied` only works on references, that's probably better here.